ScreenControl_Unity/Assets/ScreenControl/Service/ScreenControlWsBehaviour.cs

```diff
@@ -39,50 +62,36 @@ public void SendConfigurationResponse(ConfigResponse _response)
 
         protected override void OnOpen()
         {
-            // var cookies = Context.CookieCollection;
-
-            // if (cookies.Count > 0)
-            // {
-            //     string cookieApiVersion = cookies[0].Value;
-
-            //     if (cookieApiVersion != null &&
-            //             GetVersionCompability(cookieApiVersion, VersionInfo.ApiVersion) == Compatibility.COMPATIBLE)
-            //     {
-                    Debug.Log("Websocket Connection opened successfully");
-            //     }
-            //     else
-            //     {
-            //         if (cookieApiVersion == null)
-            //         {
-            //             Debug.LogError("No API version header was provided on connect!");
-            //         }
-            //         else
-            //         {
-            //             string errorMsg = $"Client API version of {cookieApiVersion} was incompatible with the Service's Core API Version of {VersionInfo.ApiVersion}";
-            //             Debug.LogError(errorMsg);
-            //             Close(CloseStatusCode.PolicyViolation, errorMsg);
-            //         }
-            //     }
-            // }
-            return;
+            Debug.Log("Websocket Connection opened");
+
+            HandshakeCompleted = false;
         }
 
         protected override void OnClose(CloseEventArgs eventArgs)
         {
             Debug.Log("Websocket Connection closed");
+
+            HandshakeCompleted = false;
         }
 
         private Compatibility GetVersionCompability(string _clientVersion, Version _coreVersion)
         {
             Version clientVersionParsed = new Version(_clientVersion);
 
-            if (clientVersionParsed.Major < _coreVersion.Major ||
-                clientVersionParsed.Minor < _coreVersion.Minor)
+            if (clientVersionParsed.Major < _coreVersion.Major)
             {
                 return Compatibility.CLIENT_OUTDATED;
             }
-            else if (clientVersionParsed.Major > _coreVersion.Major ||
-                     clientVersionParsed.Minor > _coreVersion.Minor)
+            else if (clientVersionParsed.Major > _coreVersion.Major)
+            {
+                return Compatibility.SERVICE_OUTDATED;
+            }
+
+            else if (clientVersionParsed.Minor < _coreVersion.Minor)
+            {
+                return Compatibility.CLIENT_OUTDATED;
+            }
+            else if (clientVersionParsed.Minor > _coreVersion.Minor)
             {
                 return Compatibility.SERVICE_OUTDATED;
             }
```

Comment:
Can these ifs be 'OR'-ed so we don't have duplicated returns?

Reply:
I moved from the original state to the new state because it wasn't prioritising the major versions correctly, so no, I think we need to do these individually

The issue came with comparing something like Service "0.9.1" and Client "1.0.1", because it first checks if the client is older _in either way_, it was returning an **older** client for the above because of the minor versions: 0<9